questionpy_server/web/middlewares/_authentication.py

```diff
@@ -0,0 +1,57 @@
+#  This file is part of the QuestionPy Server. (https://questionpy.org)
+#  The QuestionPy Server is free software released under terms of the MIT license. See LICENSE.md.
+#  (c) Technische Universität Berlin, innoCampus <info@isis.tu-berlin.de>
+
+from aiohttp import BasicAuth, hdrs, web
+from aiohttp.log import web_logger
+from aiohttp.typedefs import Handler
+from aiohttp.web_request import Request
+from aiohttp.web_response import StreamResponse
+
+from questionpy_server.settings import AuthSettings
+
+
+class HTTPUnauthorizedBasicAuth(web.HTTPUnauthorized):
+    """HTTP 401 Unauthorized error for BasicAuth."""
+
+    def __init__(self, reason: str | None = None):
+        super().__init__(reason=reason, headers={hdrs.WWW_AUTHENTICATE: 'Basic realm="api"'})
+
+
+def get_auth_settings(request: Request) -> AuthSettings:
+    from questionpy_server.web.app import QPyServer  # noqa: PLC0415
+
+    qpyserver = request.app[QPyServer.APP_KEY]
+    return qpyserver.settings.auth
+
+
+def get_credentials(request: Request) -> BasicAuth:
+    """Returns the BasicAuth object from the request."""
+    header = request.headers.get(hdrs.AUTHORIZATION)
+    if header is None:
+        raise HTTPUnauthorizedBasicAuth(reason="Missing authorization header")
+
+    try:
+        return BasicAuth.decode(header)
+    except ValueError as error:
+        raise HTTPUnauthorizedBasicAuth(reason="Invalid authorization header") from error
+
+
+def check_credentials(credentials: BasicAuth, users: dict[str, str]) -> bool:
+    """Checks if the given credentials are correct."""
+    return credentials.login in users and users[credentials.login] == credentials.password
```

Comment:
Ich hatte in https://github.com/questionpy-org/questionpy-server/issues/135 eigentlich vorgeschlagen, nur die Passwort-Hashes in der Config zu speichern. Das ist sicherer und hat den Vorteil, dass die Config nicht besonders geschützt werden muss. (Ich glaube Salting ist an der Stelle nicht nötig, da die Passwörter schon noch zentral vergeben werden.) Habt ihr euch bewusst dagegen entschieden oder wurde das übersehen?

Reply:
Da hast du absolut Recht, das hatte ich übersehen.

Reply:
Ich habe mal meinen PR-Kommentar geupdatet.